### PR TITLE
debug formatting

### PR DESCRIPTION
**Pull request by dsyme**
The compiler uses various records, and the default ToString fromatting used by a typical debugger ends up formatting these, which can lead to very slow debugging even when not inspecting these structures

### CODE REVIEW

**Comment by cartermp** [comment]
Just to check, this doesn't prevent inspection of these structures if it's intentional?

**Comment by dsyme** [comment]
> Just to check, this doesn't prevent inspection of these structures if it's intentional?

That's right, it's just the debug text shown next to the value, you can still expand

**Comment by dsyme** [comment]
BTW here's another `;` you might want to get rid of in your output cleanup. `{x = 1;}`

```
> {x = 1}.ToString();;
val it : string = "{x = 1;}"
```

**Comment by cartermp** [comment]
These dang semicolons

File: src/fsharp/DetupleArgs.fs [hunk 1]
```diff
@@ -644,17 +644,26 @@ let hasTransfrom penv f = Zmap.tryFind f penv.transforms
...
-      m      : Range.range }
+    { 
+      eg: TcGlobals
+
+      prefix: string
+
+      m: Range.range
+    }
+
+    override __.ToString() = "<en>"
```
Comment (brettfo) [inline src/fsharp/DetupleArgs.fs]:
```diff
-override __.ToString() = "<en>"
+override __.ToString() = "<env>"
```

**Comment by dsyme** [comment]
> why did you not use the debuggerviewattribute to change the debugger output, since that is what it is for?

Because some other formatting (e.g. F# structured formatting) ends up using ToString, and we may as well just do this formatting everywhere that strings may come out - we never want to format these records as strings, it is arbitrarily expensive and impossible to read.

> This PR appears to be almost all reformatting, with a couple of overrides added to change the debugging output. Do you think they should be split into a reformatting pr and a debug formatting pr?

No, I don't think it needs to be split, it's ok to combine a bit of reformatting with other changes that improve the codebase without changing external functionality.

**Pull request merged**